File: Api/src/Api.Domain/Entities/TaskEntity.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Domain.Entities
{
    public class TaskEntity : BaseEntity
    {
        [Required]
        [MaxLength(100)]
        public string Title { get; set; }

        [Required]
        [MaxLength(1000)]
        public string Description { get; set; }
        
        [Required]
        public bool IsCompleted { get; set; }

        public DateTime? DueDate { get; set; }

        [Required]
        public int StatusId { get; set; }

        public StatusEntity Status { get; set; }

        public int AssignedToUserId { get; set; }
        
        public UserEntity User { get; set; }

        [Required]
        public int ProjectId { get; set; }

        [JsonIgnore]
        public ProjectEntity Project { get; set; }

        [Required]
        [MaxLength(7)]
        public string Color { get; set; }
    }
}

File: App/src/app/task/entities/task.entity.ts
import { ProjectEntity } from '../../project/entities/project.entity';
import { BaseEntity } from '../../shared/entities/base.entity';
import { StatusEntity } from '../../status/entities/status.entity';
import { UserEntity } from '../../user/entities/user.entity';

export interface TaskEntity extends BaseEntity {
  title: string;
  description: string;
  isCompleted: boolean;
  dueDate?: Date;
  statusId: number;
  status?: StatusEntity;
  assignedToUserId: number;
  user?: UserEntity;
  projectId: number;
  project?: ProjectEntity;
  color: string;
}
